Add `multi_terms` aggregation type

chrispappas opened the pull request:
As mentioned in https://github.com/spatie/elasticsearch-query-builder/discussions/62

Add a `MultiTermsAggregation` that mostly follows the `TermsAggregation`, but accepts an array of field names and generates an appropriate payload.

freekmurze commented:
Tests are failing, could you look at this?

dam-bal commented on src/Aggregations/MultiTermsAggregation.php:
```diff
@@ -0,0 +1,84 @@
+        foreach($this->fields as $field) {
+            $terms[] = ['field' => $field];
+        }
+        $parameters = [
```
could you please add empty line above $parameters to make it separate from $terms building logic

dam-bal commented on tests/Aggregations/MultiTermsAggregationTest.php:
```diff
@@ -0,0 +1,46 @@
+                ],
+            ],
+        ];
+        $this->assertEquals($expectedAggArray, $aggregation->toArray());
```
ideally all assertions for test should be together, so either move them into same logic block or two separate tests maybe?

chrispappas commented:
Updated to address style comments (thank you!)

I also removed the ES Client creation and Builder creation, reduce scope down to just the new class itself. Tests pass locally using PHP 8.0.30.

```
╭─chris@chris-desktop-linux ~/src/chrispappas/elasticsearch-query-builder ‹agg-multi_terms●› 
╰─$ composer test
> vendor/bin/phpunit
PHPUnit 9.6.22 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.0.30
Configuration: /home/chris/src/chrispappas/elasticsearch-query-builder/phpunit.xml.dist
Random Seed:   1738851722
Warning:       No code coverage driver available

.....................................                             37 / 37 (100%)

Time: 00:00.012, Memory: 6.00 MB

OK (37 tests, 40 assertions)
```

bram-pkg commented on src/Aggregations/MultiTermsAggregation.php:
```diff
@@ -0,0 +1,85 @@
+    use WithMissing;
+    use WithAggregations;
+
+    /** @var string[]  */
```
```suggestion
    /** @var string[] */
```

bram-pkg commented on src/Aggregations/MultiTermsAggregation.php:
```diff
@@ -0,0 +1,85 @@
+
+    protected ?array $order = null;
+
+    /**
+     * @param  string  $name
+     * @param  string[]  $fields
+     * @return self
+     */
```
```suggestion
    /**
     * @param string[] $fields
     */
```

bram-pkg commented on src/Aggregations/MultiTermsAggregation.php:
```diff
@@ -0,0 +1,85 @@
+        return new self($name, $fields);
+    }
+
+    public function __construct(string $name, array $fields)
```
```suggestion
    /**
     * @param string[] $fields
     */
    public function __construct(string $name, array $fields)
```

bram-pkg commented on src/Aggregations/MultiTermsAggregation.php:
```diff
@@ -0,0 +1,85 @@
+    public function payload(): array
+    {
+        $terms = [];
+        foreach($this->fields as $field) {
```
```suggestion

        foreach($this->fields as $field) {
```

chrispappas commented:
Fixed the style issues. Also thanks for maintaining FakerPHP @bram-pkg, very useful and have been using it for a long time now since the fzaninotto days.

FYI, I think this is unrelated to my branch specifically (tested on origin/main as well), but the `composer format` command is failing:

```
╭─chris@chris-desktop-linux ~/src/chrispappas/elasticsearch-query-builder ‹main› 
╰─$ composer format -vv                                                                                                                                                                                                                                                                                  1 ↵
> format: vendor/bin/php-cs-fixer fix --allow-risky=yes
You are running PHP CS Fixer v2, which is not maintained anymore. Please update to v3.
You may find an UPGRADE guide at https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.3.0/UPGRADE-v3.md .
If you need help while solving warnings, ask at https://gitter.im/PHP-CS-Fixer, we will help you!

PHP CS Fixer 2.19.3 Testament by Fabien Potencier and Dariusz Ruminski
Runtime: PHP 8.0.30
Loaded config default.
Using cache file ".php_cs.cache".

In Finder.php line 614:
                                                                                 
  [LogicException]                                                               
  You must call one of in() or append() methods before iterating over a Finder.  
                                                                                 

Exception trace:
  at /home/chris/src/chrispappas/elasticsearch-query-builder/vendor/symfony/finder/Finder.php:614
 Symfony\Component\Finder\Finder->getIterator() at /home/chris/src/chrispappas/elasticsearch-query-builder/vendor/friendsofphp/php-cs-fixer/src/Runner/Runner.php:120
 PhpCsFixer\Runner\Runner->fix() at /home/chris/src/chrispappas/elasticsearch-query-builder/vendor/friendsofphp/php-cs-fixer/src/Console/Command/FixCommand.php:228
 PhpCsFixer\Console\Command\FixCommand->execute() at /home/chris/src/chrispappas/elasticsearch-query-builder/vendor/symfony/console/Command/Command.php:256
 Symfony\Component\Console\Command\Command->run() at /home/chris/src/chrispappas/elasticsearch-query-builder/vendor/symfony/console/Application.php:971
 Symfony\Component\Console\Application->doRunCommand() at /home/chris/src/chrispappas/elasticsearch-query-builder/vendor/symfony/console/Application.php:290
 Symfony\Component\Console\Application->doRun() at /home/chris/src/chrispappas/elasticsearch-query-builder/vendor/friendsofphp/php-cs-fixer/src/Console/Application.php:102
 PhpCsFixer\Console\Application->doRun() at /home/chris/src/chrispappas/elasticsearch-query-builder/vendor/symfony/console/Application.php:166
 Symfony\Component\Console\Application->run() at /home/chris/src/chrispappas/elasticsearch-query-builder/vendor/friendsofphp/php-cs-fixer/php-cs-fixer:115
 include() at /home/chris/src/chrispappas/elasticsearch-query-builder/vendor/bin/php-cs-fixer:119

fix [--path-mode PATH-MODE] [--allow-risky ALLOW-RISKY] [--config CONFIG] [--dry-run] [--rules RULES] [--using-cache USING-CACHE] [--cache-file CACHE-FILE] [--diff] [--diff-format DIFF-FORMAT] [--format FORMAT] [--stop-on-violation] [--show-progress SHOW-PROGRESS] [--] [<path>...]

Script vendor/bin/php-cs-fixer fix --allow-risky=yes handling the format event returned with error code 1
```

bram-pkg commented on src/Aggregations/MultiTermsAggregation.php:
```diff
@@ -0,0 +1,87 @@
+        return $this;
+    }
+
+    public function order(array $order): self
```
I think it would make sense to add support for the `Sorting` interface that is included in this package.

An example of that would be:

```suggestion
    /**
     * @param Sorting|Sorting[]|null $sort
     */
    public function order(array $order): self
    {
        $this->sort = is_array($sort) ? $sort : [$sort];
```

And then down below, an `array_map` like this:


```php
        if ($this->sort) {
            $parameters['sort'] = array_map(fn (Sorting $sort) => $sort->toArray(), $this->sort);
        }
```

chrispappas replied:
I will add support for both order AND size, which I should have included from the start, thanks!

bram-pkg commented:
> Updated to address style comments (thank you!)
> 
> I also removed the ES Client creation and Builder creation, reduce scope down to just the new class itself. Tests pass locally using PHP 8.0.30.
> 
> ```
> ╭─chris@chris-desktop-linux ~/src/chrispappas/elasticsearch-query-builder ‹agg-multi_terms●› 
> ╰─$ composer test
> > vendor/bin/phpunit
> PHPUnit 9.6.22 by Sebastian Bergmann and contributors.
> 
> Runtime:       PHP 8.0.30
> Configuration: /home/chris/src/chrispappas/elasticsearch-query-builder/phpunit.xml.dist
> Random Seed:   1738851722
> Warning:       No code coverage driver available
> 
> .....................................                             37 / 37 (100%)
> 
> Time: 00:00.012, Memory: 6.00 MB
> 
> OK (37 tests, 40 assertions)
> ```

It indeed looks like this is unrelated.. I can take a look at modernising our PHP CS Fixer configuration..

chrispappas commented:
As I am washing the dishes thinking about this... I am thinking this still needs some work. Realizing that we need to support a simple array being supplied to the order method to maintain API compat with the other agg type. More to come soon, will update when it's ready for another review! Thanks for everyone's attention so far.
